solutions/task_5.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python3
+"""
+TASK: Implement at least functions 'test1_func1', 'test2_func1', 'test3_func1', 'test3_func2'
+to execute unittests without errors.
+"""
+import functools
+import random
+import time
+import types
+import unittest
+
+MY_DICT = {
+    'a': [1, 2, 3],
+    'b': {
+        'c': 1,
+        'd': {'e': 'f', 'f': (5, 6, 7)},
+        'f': [7, [8, [9, 10]]],
+        'i': 'hello',
+    },
+}
+EXPECTED_RESULT = [1, 2, 3, 'hello', 1, 'f', 5, 6, 7, 7, 8, 9, 10]
+
+
+def long_func(multiplier, int_value):
+    time.sleep(1)
+    return multiplier * int_value
+
+
+def test1_func1(value, output=[]):
+    if isinstance(value, dict):
+        for item in value.values():
+            test1_func1(item, output)
+    elif isinstance(value, (list, tuple)):
+        for item in value:
+            test1_func1(item, output)
+    else:
+        output.append(value)
+    return output
+
+
+def test2_func1(func, ints):
+    for value in ints:
+        yield func(value)
+
+
+def test3_func1(iterable):
+    if isinstance(iterable, dict):
+        for value in iterable.values():
+            gen = test3_func1(value)
+            for item in gen:
+                yield item
+    else:
+        yield iterable
+
+
+def test3_func2(iterable):
+    if isinstance(iterable, (list, tuple, types.GeneratorType)):
```

Comment:
Why do you need ```types.GeneratorType```?

Reply:
Because the output of test3_func1 is generator and when I pass this output to test3_func2 I would like this if to be true as well as for each list or tuple passed to the same function within recursion.